common/work.c

```diff
@@ -17,45 +17,88 @@ static struct {
 	void         *param;
 } work_items[MAX_WORK_ITEMS];
 
+static size_t queue_head = 0;
+static size_t queue_tail = MAX_WORK_ITEMS;
+
+/**
+ * Determines if the queue is full using the given arguments. The values of
+ * the head and tail range from 0 to 2 * MAX_WORK_ITEMS. This allows us to
+ * distinguish between a full queue or an empty queue if the head is equal to
+ * the tail or if the head and tail are exactly MAX_WORK_ITEMS elements apart,
+ * respectively.
+ *
+ * @param  head The index of the head of the work_items queue
+ * @param  tail The index of the tail of the work_items queue
+ * @return      True if the queue is full, otherwise false
+ */
+static inline bool
+queue_full(size_t head, size_t tail)
+{
+	if (head == tail)
+		return true;
+	return false;
+}
+
+/**
+ * Determines if the queue is empty using the given arguments.
+ *
+ * @param  head The index of the head of the work_items queue
+ * @param  tail The index of the tail of the work_items queue
+ * @return      True if the queue is empty, otherwise false
+ */
+static inline bool
+queue_empty(size_t head, size_t tail)
+{
+	if ((head + MAX_WORK_ITEMS) % (2 * MAX_WORK_ITEMS) == tail)
```

Comment:
Same thing here, you can remove the `if` and `true`/`false` lines, and return the boolean directly.